Assets/MRTK/Core/Providers/BaseController.cs

```diff
@@ -96,17 +102,58 @@ protected BaseController(TrackingState trackingState, Handedness controllerHande
         /// <summary>
         /// The default interactions for this controller.
         /// </summary>
-        public virtual MixedRealityInteractionMapping[] DefaultInteractions { get; } = null;
+        public virtual MixedRealityInteractionMapping[] DefaultInteractions
+        {
+            get
+            {
+                System.Collections.Generic.IReadOnlyList<MixedRealityInputActionMapping> definitionInteractions = Definition?.GetDefaultMappings(ControllerHandedness);
+                MixedRealityInteractionMapping[] defaultInteractions = new MixedRealityInteractionMapping[definitionInteractions.Count];
+                for (int i = 0; i < definitionInteractions.Count; i++)
+                {
+                    defaultInteractions[i] = new MixedRealityInteractionMapping((uint)i, definitionInteractions[i]);
+                }
+                return defaultInteractions;
+            }
+        }
 
         /// <summary>
-        /// The Default Left Handed interactions for this controller.
+        /// The default left-handed interactions for this controller.
         /// </summary>
-        public virtual MixedRealityInteractionMapping[] DefaultLeftHandedInteractions { get; } = null;
+        public virtual MixedRealityInteractionMapping[] DefaultLeftHandedInteractions
+        {
+            get
```

Comment:
1. Do you think it might be a good idea to extract a function from here as the left and right interactions share a similar logic?
2. Do you foresee this function being called often? I am not sure if the performance impact is negligible or not.

Reply:
1. Ahh good call, yeah!
2. It shouldn't be called at all at runtime. This is typically only used to create the initial mapping profile. After that, the mappings at runtime should be loaded out of the profile instead of here.

Reply:
1. Refactored out `BuildInteractions`. Thanks for the suggestion!